### PR TITLE
feature: support openQASM serialization of v gate

### PR DESCRIPTION
**Pull request by valegogu**
*Issue #, if available:*
#329 
*Description of changes:*
Added support for openQASM serialization of v gate.

*Testing done:*
Added/ran unit tests with 88% coverage.
Created/ran test script below as an integration test.
```
circuit = Circuit().v(0)
print(f"{circuit.to_ir(ir_type=IRType.OPENQASM).source}")
d = AwsDevice("arn:aws:braket:::device/quantum-simulator/amazon/sv1")
task = d.run(circuit.to_ir(ir_type=IRType.OPENQASM), shots=10)
print(f"{task.id}")
res = task.result()
print(res)
```

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [ x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/circuits/test_gates.py [hunk 1]
```diff
@@ -354,6 +354,30 @@ def test_ir_gate_level(testclass, subroutine_name, irclass, irsubclasses, kwargs
...
+            [4],
+            OpenQASMSerializationProperties(qubit_reference_type=QubitReferenceType.VIRTUAL),
+            "h q[4];",
+        ),
+        (
+            Gate.H(),
+            [4],
+            OpenQASMSerializationProperties(qubit_reference_type=QubitReferenceType.PHYSICAL),
+            "h $4;",
+        ),
```
Comment (ajberdy) [inline test/unit_tests/braket/circuits/test_gates.py:368]: something strange is happening with the diff here. Did you branch off of main or off of your previous h-gate branch?

Comment (ajberdy) [reply test/unit_tests/braket/circuits/test_gates.py:368]: Oh actually this isn't a diff thing. It looks like there are new H gate test cases being added here in addition to the ones above

Comment (valegogu) [reply test/unit_tests/braket/circuits/test_gates.py:368]: Oh, I merged before starting on the v branch, and the H gate test cases were already there. Should I do something to change that?

Comment (valegogu) [reply test/unit_tests/braket/circuits/test_gates.py]: I've removed the extra H gate test case

**Review by ajberdy — APPROVED** [review]
Sweet!

**Pull request merged**